src/server/frontend_xwayland/xwayland_server.cpp

```diff
@@ -213,16 +240,33 @@ void mf::XWaylandServer::execl_xwayland(int wl_client_client_fd, int wm_client_f
     action.sa_handler = SIG_IGN;
     sigaction(SIGUSR1, &action, nullptr);
 
-    execl(
-        xwayland_path.c_str(),
-        xwayland_path.c_str(),
-        dsp_str.c_str(),
-        "-rootless",
-        "-listen", abstract_socket_fd_str.c_str(),
-        "-listen", socket_fd_str.c_str(),
-        "-wm", wm_fd_str.c_str(),
-        "-terminate",
-        NULL);
+    std::vector<char const*> args =
+        {
+            xwayland_path.c_str(),
+            dsp_str.c_str(),
+            "-rootless",
+            "-wm", wm_fd_str.c_str(),
+            "-terminate",
+        };
+
+    if (sockets.abstract_socket_fd != -1)
+    {
+        args.push_back("-listen");
+        args.push_back(strdup(std::to_string(sockets.abstract_socket_fd).c_str()));
```

Comment:
This leaks, right? Can we easily make it not leak the string?

Reply:
Comment added